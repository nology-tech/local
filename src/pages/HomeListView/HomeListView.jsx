import "./HomeListView.scss";
import Card from "../../components/Card/Card";
import Layout from "../../components/Layout/Layout";
import Filter from "../../components/Filter/Filter";
import { useContext, useState } from "react";
import UserContext from "../../context/UserContext";
import TextField from "../../components/TextField/TextField";
import MapIcon from "../../assets/icons/mapMarkerIcon.svg";
import Button from "../../components/Button/Button";
import { Link } from "react-router-dom";
import businessArr from "../Home/Home";
import handleSaveToFavourites from "../Home/Home";
const HomeListView = () => {
  const { user } = useContext(UserContext);
  if (user === null) {
    return <p>loading...</p>;
  }
  const [homeBusinessArray, sethomeBusinessArray] = useState(businessArr);

  return (
    <Layout isWithMenu={true}>
      <div className="home-list-view__search--bar-container">
        <img
          className="home-list-view__map-icon"
          src={MapIcon}
          type="image/svg+xml"
          alt="an icon of a map marker"
        />

        <TextField
          uniqueId="location-search-bar"
          inputType="text"
          modifier="location-search-bar"
          placeholderText="e.g. CH451HE"
        />

        <Link to="../*" className="home-list-view__link--map-view">
          <Button buttonName="map-view" buttonText="Map View" />
        </Link>
      </div>

      <main className="home-list-view__container">
        <Filter
          favArray={homeBusinessArray}
          setAllCardsArr={sethomeBusinessArray}
        />
        <Card
          cardArray={homeBusinessArray}
          page="home-list-view"
          primaryButtonOnClick={handleSaveToFavourites}
        />
      </main>
    </Layout>
  );
};
export default HomeListView;
