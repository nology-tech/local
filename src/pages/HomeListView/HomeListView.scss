@use "../../styles/variables" as *;

.home-list-view__search--bar-container {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  position: relative;
  width: 100%;
  .home-list-view__map-icon {
    position: absolute;
    height: 2.5rem;
    margin-left: 1rem;
    z-index: 2;
  }
}
.text-field--location-search-bar {
  display: block;
}
.text-field__label--location-search-bar {
  display: none;
}
.text-field__input--location-search-bar {
  width: 100%;
  text-align: center;
  &::placeholder {
    color: $text-color;
    text-align: center;
    font-weight: 600;
  }
}
.home-list-view__container {
  display: flex;
  flex-flow: column nowrap;
  align-items: stretch;
  padding: 0 1rem;
}
