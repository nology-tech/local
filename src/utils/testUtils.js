import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
