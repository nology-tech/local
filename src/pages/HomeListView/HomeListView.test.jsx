import { customRender } from "../../utils/testUtils";
import HomeListView from "./HomeListView";

it("should render the text on the display", () => {
  const { container } = customRender(<HomeListView />);

  expect(container).toMatchSnapshot();
});
