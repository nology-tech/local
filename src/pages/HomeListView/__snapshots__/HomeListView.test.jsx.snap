// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text on the display 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="header"
    >
      <img
        alt=""
        src="BackButton.svg"
      />
      <div
        class="header__container"
      >
        <div
          class="header__logo-title"
        >
          <img
            alt=""
            src="AppLogo.svg"
          />
          <h1
            class="header__title-local"
          >
            LOCAL
          </h1>
        </div>
      </div>
      <div
        class="menu-icon__container"
        role="button"
      >
        <svg
          alt="Menu-icon"
          class="menu-icon__icon"
          fill="none"
          height="30"
          viewBox="0 0 44 30"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 30H44V25H0V30ZM0 17.5H44V12.5H0V17.5ZM0 0V5H44V0H22H0Z"
            fill="black"
          />
        </svg>
      </div>
    </div>
    <div
      class="layout__menu menu-closed"
    >
      <div
        class="menu"
      >
        <div
          class="menu__header"
        >
          <img
            alt="App Logo"
            class="menu__app-logo"
            src="logo.svg"
          />
          <h1
            class="menu__app-title"
          >
            LOCAL
          </h1>
          <img
            alt="Close Button"
            class="menu__close"
            src="close-btn.svg"
          />
        </div>
        <div
          class="menu__home"
        >
          <a
            class="menu__link"
            href="/home"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Home Logo"
                class="menu__logo"
                src="home.svg"
              />
              <h2
                class="menu__option"
              >
                Home
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__favourites"
        >
          <a
            class="menu__link"
            href="/favourites"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Home Logo"
                class="menu__logo"
                src="favourites.svg"
              />
              <h2
                class="menu__option"
              >
                Favourites
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__info"
        >
          <a
            class="menu__link"
            href="/account"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Account Logo"
                class="menu__logo"
                src="account.svg"
              />
              <h2
                class="menu__option"
              >
                Manage Account
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__about"
        >
          <a
            class="menu__link"
            href="/about"
          >
            <div
              class="menu__page"
            >
              <img
                alt="About Logo"
                class="menu__logo"
                src="about.svg"
              />
              <h2
                class="menu__option"
              >
                About
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__contact"
        >
          <a
            class="menu__link"
            href="/contact"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Contact Logo"
                class="menu__logo"
                src="contact.svg"
              />
              <h2
                class="menu__option"
              >
                Contact
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="home-list-view__search--bar-container"
    >
      <img
        alt="an icon of a map marker"
        class="home-list-view__map-icon"
        src="mapMarkerIcon.svg"
        type="image/svg+xml"
      />
      <div
        class="text-field text-field--location-search-bar"
      >
        <label
          class="text-field__label text-field__label--location-search-bar"
          for="location-search-bar"
        />
        <input
          class="text-field__input text-field__input--location-search-bar"
          id="location-search-bar"
          placeholder="e.g. CH451HE"
          type="text"
        />
      </div>
      <a
        class="home-list-view__link--map-view"
        href="/*"
      >
        <button
          class="button button--map-view"
        >
          Map View
        </button>
      </a>
    </div>
    <main
      class="home-list-view__container"
    >
      <div
        class="filter"
      >
        <button
          class="button button--filter"
        >
          FOOD & DRINK
        </button>
        <button
          class="button button--filter"
        >
          CULTURE
        </button>
        <button
          class="button button--filter"
        >
          OTHER
        </button>
      </div>
      <div
        class="card card--home-list-view"
      >
        <div
          class="card__header-container"
        >
          <h2
            class="card__heading card__heading--food"
          >
            Medicine Bakery
          </h2>
          <div
            class="card__image-container"
          >
            <img
              alt=""
              class="card__image"
              src="forkIcon.svg"
            />
          </div>
        </div>
        <p
          class="card__text"
        >
          Medicine Bakery is a small, family-run bakery located in the city of Birmingham.
        </p>
        <div
          class="card__buttons-container"
        >
          <button
            class="button button--details"
          >
            Details
          </button>
          <button
            class="button button--save"
          >
            Save
          </button>
        </div>
      </div>
      <div
        class="card card--home-list-view"
      >
        <div
          class="card__header-container"
        >
          <h2
            class="card__heading card__heading--culture"
          >
            La M C
          </h2>
          <div
            class="card__image-container"
          >
            <img
              alt=""
              class="card__image"
              src="paintIcon.svg"
            />
          </div>
        </div>
        <p
          class="card__text"
        >
          No description, try it works.com
        </p>
        <div
          class="card__buttons-container"
        >
          <button
            class="button button--details"
          >
            Details
          </button>
          <button
            class="button button--save"
          >
            Save
          </button>
        </div>
      </div>
      <div
        class="card card--home-list-view"
      >
        <div
          class="card__header-container"
        >
          <h2
            class="card__heading card__heading--food"
          >
            The Indian Streatery
          </h2>
          <div
            class="card__image-container"
          >
            <img
              alt=""
              class="card__image"
              src="forkIcon.svg"
            />
          </div>
        </div>
        <p
          class="card__text"
        >
          A restaurant and Indian restaurant known for its delicious food and authentic flavors.
        </p>
        <div
          class="card__buttons-container"
        >
          <button
            class="button button--details"
          >
            Details
          </button>
          <button
            class="button button--save"
          >
            Save
          </button>
        </div>
      </div>
      <div
        class="card card--home-list-view"
      >
        <div
          class="card__header-container"
        >
          <h2
            class="card__heading card__heading--food"
          >
            The Lost & Found Birmingham
          </h2>
          <div
            class="card__image-container"
          >
            <img
              alt=""
              class="card__image"
              src="forkIcon.svg"
            />
          </div>
        </div>
        <p
          class="card__text"
        >
          A pub, bar, and more, offering a wide range of drinks and a fun atmosphere.
        </p>
        <div
          class="card__buttons-container"
        >
          <button
            class="button button--details"
          >
            Details
          </button>
          <button
            class="button button--save"
          >
            Save
          </button>
        </div>
      </div>
      <div
        class="card card--home-list-view"
      >
        <div
          class="card__header-container"
        >
          <h2
            class="card__heading card__heading--food"
          >
            Salt Dog Slim's
          </h2>
          <div
            class="card__image-container"
          >
            <img
              alt=""
              class="card__image"
              src="forkIcon.svg"
            />
          </div>
        </div>
        <p
          class="card__text"
        >
          This is a lively bar and restaurant that serves a variety of beers, wines, and cocktails, as well as classic pub food like burgers, sandwiches, and wings.
        </p>
        <div
          class="card__buttons-container"
        >
          <button
            class="button button--details"
          >
            Details
          </button>
          <button
            class="button button--save"
          >
            Save
          </button>
        </div>
      </div>
      <div
        class="card card--home-list-view"
      >
        <div
          class="card__header-container"
        >
          <h2
            class="card__heading card__heading--other"
          >
            Scouse Bird Shop
          </h2>
          <div
            class="card__image-container"
          >
            <img
              alt=""
              class="card__image"
              src="other.svg"
            />
          </div>
        </div>
        <p
          class="card__text"
        >
          A fun and quirky shop featuring a range of products and gifts inspired by local culture and humor.
        </p>
        <div
          class="card__buttons-container"
        >
          <button
            class="button button--details"
          >
            Details
          </button>
          <button
            class="button button--save"
          >
            Save
          </button>
        </div>
      </div>
    </main>
  </div>
</div>
`;
